src/Workspaces/Core/Portable/Workspace/WorkspaceEventRegistration.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace Microsoft.CodeAnalysis;
+
+internal sealed class WorkspaceEventRegistration(Action unregisterAction) : IDisposable
```

Comment:
possible fro this to be a struct?

Reply:
Anythings possible! But, I don't know if it's desirable. I would expect most users to interact with this as a stored IDisposable, which means boxing if this is a struct. Also, this is only allocated during event registration, so not the highest of traffic paths.